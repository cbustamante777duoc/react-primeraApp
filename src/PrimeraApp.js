//import React, { Fragment } from 'react';
import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({saludo, subtitulo='valor por defecto',equipo}) =>{

  //  const saludar = 'hola ';
    /*const saludo = {
        nombre:'Rosie',
        edad:23
    }*/

    return (
        <>
            <h1>{saludo}</h1>
            {/* <pre>JSON.stringify(saludo, null, 3)</pre> */}
            <p>{subtitulo}</p>
            <hr></hr>
            <p>{equipo}</p>
        </>
    );
            
}

//definir que tipo de variable va a recibir el componente
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
    subtitulo: PropTypes.string,
   
}

//valor por defecto que va a tener el componente
PrimeraApp.defaultProps = {
    equipo:'Real madrid'
}

export default PrimeraApp;