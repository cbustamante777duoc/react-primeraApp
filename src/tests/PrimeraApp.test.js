import React from 'react';
import PrimeraApp from "../PrimeraApp";
import {shallow} from 'enzyme';

describe('Pruebas en <PrimeraApp/>', () => {
    // test('debe de mostrar el mensaje "hola, jennie"', () => {

    //     const saludo = 'hola, jennie';
    //     //destruturacion
    //     const {getByText} = render(<PrimeraApp saludo={saludo}/>);
    //     //pregunta si el contenido de la variable saludo se encuentra en html
    //     expect(getByText(saludo )).toBeInTheDocument();

    // })
    test('debe de mostrar la <PrimeraApp/> correctamente', () => {
        const saludo = 'hola, jennie';
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>)

        expect(wrapper).toMatchSnapshot();

    })
    
    
})
