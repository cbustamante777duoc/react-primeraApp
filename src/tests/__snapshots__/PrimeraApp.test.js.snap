// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> debe de mostrar la <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    hola, jennie
  </h1>
  <p>
    valor por defecto
  </p>
  <hr />
  <p>
    Real madrid
  </p>
</Fragment>
`;
